src/rpc/blockchain.cpp

```diff
@@ -2771,7 +2771,15 @@ static RPCHelpMan loadtxoutset()
     }
 
     SnapshotMetadata metadata;
-    afile >> metadata;
+    try {
+        // Read data from the file
+        afile >> metadata;
+    } catch (const std::exception&) {
+        // Handle any exception, which includes reaching the end of the file e.g. file size < sizeOf(metadata)
+        throw JSONRPCError(RPC_INTERNAL_ERROR, strprintf("Unable to load UTXO snapshot, "
```

Comment:
Shouldn't use `RPC_INTERNAL_ERROR`. As per it description: "this error should only be used for genuine errors in bitcoind (for example datadir corruption)"

Maybe use `RPC_DESERIALIZATION_ERROR` or `RPC_MISC_ERROR.

Reply:
It makes sense, I'll fix it, thanks!

Reply:
done!